src/Microsoft.ML.TimeSeries/ExtensionsCatalog.cs

```diff
@@ -145,5 +147,37 @@ public static class TimeSeriesCatalog
         public static SrCnnAnomalyEstimator DetectAnomalyBySrCnn(this TransformsCatalog catalog, string outputColumnName, string inputColumnName,
             int windowSize=64, int backAddWindowSize=5, int lookaheadWindowSize=5, int averageingWindowSize=3, int judgementWindowSize=21, double threshold=0.3)
             => new SrCnnAnomalyEstimator(CatalogUtils.GetEnvironment(catalog), outputColumnName, windowSize, backAddWindowSize, lookaheadWindowSize, averageingWindowSize, judgementWindowSize, threshold, inputColumnName);
+
+        /// <summary>
+        /// Singular Spectrum Analysis (SSA) model for modeling univariate time-series.
+        /// For the details of the model, refer to http://arxiv.org/pdf/1206.6910.pdf.
+        /// </summary>
+        /// <param name="catalog">Catalog.</param>
+        /// <param name="inputColumnName">The name of the column on which forecasting needs to be performed.</param>
+        /// <param name="trainSize">The length of series from the begining used for training.</param>
+        /// <param name="seriesLength">The length of series that is kept in buffer for modeling (parameter N).</param>
+        /// <param name="windowSize">The length of the window on the series for building the trajectory matrix (parameter L).</param>
+        /// <param name="discountFactor">The discount factor in [0,1] used for online updates (default = 1).</param>
+        /// <param name="rankSelectionMethod">The rank selection method (default = Exact).</param>
+        /// <param name="rank">The desired rank of the subspace used for SSA projection (parameter r). This parameter should be in the range in [1, windowSize].
+        /// If set to null, the rank is automatically determined based on prediction error minimization. (default = null)</param>
+        /// <param name="maxRank">The maximum rank considered during the rank selection process. If not provided (i.e. set to null), it is set to windowSize - 1.</param>
+        /// <param name="shouldComputeForecastIntervals">The flag determining whether the confidence bounds for the point forecasts should be computed. (default = true)</param>
+        /// <param name="shouldstablize">The flag determining whether the model should be stabilized.</param>
+        /// <param name="shouldMaintainInfo">The flag determining whether the meta information for the model needs to be maintained.</param>
+        /// <param name="maxGrowth">The maximum growth on the exponential trend</param>
```

Comment:
Is this parameter needed by user?

Reply:
Its an optional parameter.

---
In reply to: [289948356](https://github.com/dotnet/machinelearning/pull/1900#discussion_r289948356) [](ancestors = 289948356)